modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/net/jms/Acknowledge.java

```diff
@@ -48,14 +51,33 @@
 @BallerinaAnnotation(annotationName = "Param", attributes = { @Attribute(name = "deliveryStatus",
         value = "Specify whether message delivery is SUCCESS or ERROR") })
 public class Acknowledge extends AbstractNativeFunction {
+    private static final Logger log = LoggerFactory.getLogger(Acknowledge.class);
+
     public BValue[] execute(Context ctx) {
         BMessage msg = (BMessage) getArgument(ctx, 0);
         CarbonMessage carbonMessage = msg.value();
         String deliveryStatus = getArgument(ctx, 1).stringValue();
+        Object jmsSessionAcknowledgementMode = carbonMessage
+                .getProperty(JMSConstants.JMS_SESSION_ACKNOWLEDGEMENT_MODE);
 
-        if (ctx.getBalCallback() != null) {
-            carbonMessage.setProperty(JMSConstants.JMS_MESSAGE_DELIVERY_STATUS, deliveryStatus);
-            ctx.getBalCallback().done(carbonMessage);
+        if (null == jmsSessionAcknowledgementMode) {
+            log.warn("JMS Acknowledge function can only be used with JMS Messages. "
```

Comment:
This property is not optional for jms messages, as the relevant property will be added for all the jms messages in the transport level. For other message types, if some user has mistakenly tried this, I think we need to give some level of indication on that this related to jms. WDYT?